rplugin/python3/molten/outputbuffer.py

```diff
@@ -292,6 +292,7 @@ def show_floating_win(self, anchor: Position) -> None:
 
         # Clear buffer:
         self.nvim.funcs.deletebufline(self.display_buf.number, 1, "$")
+        self.canvas.clear()
```

Comment:
this line causes images that are part of virtual text output to disappear when I open the output in a floating window.

Reply:
Virtual output was always broken for me, so I assumed this was by design/not a priority. To have well-defined behavior between virtual output and floating windows, I assume a separate canvas would be necessary for each?

Reply:
Currently I'm able to show images as virtual text, open a popup, and see the image there as well. The image doesn't disappear when I close the output (although sometimes image.nvim bugs and doesn't render the image correctly until I scroll).

You can open an issue for virtual text if it's not working for you.